src/main/java/org/springframework/data/elasticsearch/SpringDataElasticsearchException.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch;
+
+import org.springframework.dao.UncategorizedDataAccessException;
+
+/**
+ * @author Peter-Josef Meisch
+ * @since 4.0
+ */
+public class SpringDataElasticsearchException extends UncategorizedDataAccessException {
```

Comment:
Instead of adding another exception type, let's throw `UncategorizedDataAccessException` in the places that we're not fully sure about the cause.

Reply:
`UncategorizedDataAccessException` is abstract, that's why I introduced this class

Reply:
That intent wasn’t obvious from the name and usage. Then let’s rename it to UncategorizedElasticaearchException for a clearer picture.